### DISCUSSION TITLE
UmbracoMemberAuthorizeAttribute optional parameter

### DISCUSSION THREAD

**Discussion by Sander-Los** [umbraco/Umbraco-CMS#13790] [Features and ideas]
Hi,

In the UmbracoMemberAuthorizeAttribute there is a posibility to allow membertype/membergroup/specific members.

When I want to apply this attribute to a controller, I have to specificly supply all 3 parameter while I for Example only want to provide membergroups. 

Example from the documentation listed below:"

`[UmbracoMemberAuthorize("", "VIP", "")]`

Would it not be nicer to have the parameters being optional by setting a default value?

```
 public UmbracoMemberAuthorizeAttribute(string allowType = string.Empty, string allowGroup = string.Empty, string allowMembers= string.Empty)
        : base(typeof(UmbracoMemberAuthorizeFilter)) => Arguments = new object[] { allowType, allowGroup, allowMembers };
}
```
that way, we can specify what we want to provide:

`[UmbracoMemberAuthorize(allowgroups:"VIP")]`

I am curious what you thought about this.

https://docs.umbraco.com/v/10.x-lts/umbraco-cms/implementation/controllers#members-and-front-end-authorization
https://docs.umbraco.com/v/10.x-lts/umbraco-cms/reference/routing/umbraco-api-controllers/authorization#using-memberauthorizeattribute

**Comment by bergmania** [comment]
Sounds like a good idea, but If im not wrong, this is a binary breaking change?

At least it needs to be tested before we can introduce such a change.
